src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1953,6 +1977,21 @@ protected void ExecuteIndexing()
             }
         }
 
+        private TimeSpan UpdateElapsedTimeFromLastQuery(DateTime now)
```

Comment:
Please consider that it can be called concurrently with `MarkQueried`. This can potentially lead to skipping / overwriting some real elapsed querying time.

Reply:
Handled.